streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java

```diff
@@ -580,6 +584,36 @@ public List<String> historyUploadJars() {
         .collect(Collectors.toList());
   }
 
+  @Override
+  public String k8sStartLog(Long id, Integer offset, Integer limit) throws Exception {
+    Application application = getById(id);
+    AssertUtils.state(application != null);
+    if (ExecutionMode.isKubernetesMode(application.getExecutionModeEnum())) {
+      CompletionStage<String> stage =
+          CompletableFuture.supplyAsync(
+                  () ->
+                      KubernetesDeploymentHelper.watchDeploymentLog(
+                          application.getK8sNamespace(),
+                          application.getJobName(),
+                          application.getJobId()))
+              .exceptionally(
+                  e -> {
+                    try {
+                      return String.format(
+                          "%s/%s_err.log", WebUtils.getAppTempDir(), application.getJobId());
+                    } catch (Exception ex) {
+                      throw new ApiDetailException(
+                          "Generate log path exception: " + ex.getMessage());
+                    }
+                  })
+              .thenApply(path -> logClient.rollViewLog(String.valueOf(path), offset, limit));
+      CompletableFuture<String> future = stage.toCompletableFuture();
+      return future.get(5, TimeUnit.SECONDS);
```

Comment:
Should we cancel the future after timeout?

Reply:
> Should we cancel the future after timeout?

Thanks for your review, this code has been improved and some bugs have been solved, usually the feature does not need to be closed manually.